### PR TITLE
Allow DSpace's ConfigurationService to override Spring Boot properties

### PR DESCRIPTION
**Pull request by tdonohue**
## References
* Enhancement to #2729 (which ensured our Spring Boot settings had a higher `max-file-size` set for uploads)

## Description
When reviewing #2729, I realized that some Spring Boot properties would be nice to be able to override in `local.cfg`. The `max-file-size` is a good example, as some sites may want to increase or decrease this easily (without rebuilding the entire Server webapp). 

This small PR does that by ensuring the DSpace ConfigurationService is checked *first* for properties.  See the small, one-line change to our `DSpaceConfigurationInitializer`.

## Instructions for Reviewers
An easy way to test this is to simply decrease the Spring Boot max upload size in your `local.cfg` and see if you can reproduce the bug that was fixed by #2729.

In other words, this setting in your `local.cfg` will ensure you cannot upload any file >1MB.
```
spring.servlet.multipart.max-file-size = 1MB
```
Removing that setting from your `local.cfg` will then result in files up to 512MB being allowable (as that's the default now in our Spring Boot `application.properties` file).

### CODE REVIEW

**Review by benbosman — APPROVED** [review]
Thanks for the updates @tdonohue

**Comment by tdonohue** [comment]
Merging as this is a small one-line change (along with a lot of comments/examples in configs) and has another review.  I've tested this locally (via the instructions above) and found it to be working. All tests also succeed.
